lib/matplotlib/axes/_axes.py

```diff
@@ -1868,25 +1868,50 @@ def step(self, x, y, *args, **kwargs):
 
         return self.plot(x, y, *args, **kwargs)
 
-    @_preprocess_data(replace_names=["left", "height", "width", "bottom",
+    @_preprocess_data(replace_names=["x", "left",
+                                     "height", "width",
+                                     "y", "bottom",
                                      "color", "edgecolor", "linewidth",
                                      "tick_label", "xerr", "yerr",
                                      "ecolor"],
-                      label_namer=None)
+                      label_namer=None,
+                      positional_parameter_names=['x',
```

Comment:
Why do we need this?  (I'ml not too familiar with the exact semantics of _preprocess_data, but at least `y` looks a bit fishy in the argument list here)

Reply:
the 'y' is junk from a second rename that I ended up walking back.  The purpose of this list is to tell the pre-processor the name of input hidden in the '*args' so it know which entries to try and look up in `data`, however in this case we want to do all of them so there is a simpler way.